wgsl/index.bs

```diff
@@ -8605,11 +8624,192 @@ The following algorithm describes how to compute these tags for a given function
+notations:
+* *Vin*(*S*) is the value of `x` prior the execution of the statement *S*.
+* *Vout*(*S*) is the value of `x` after the execution of the statement *S*.
+* *Vout*(*prev*) is the value of `x` prior to the execution of the current statement.
+* *Vin*(*next*) is the value of `x` prior to the execution of the next statement.
+* *V*(*e*) is a value node for an expression as in the subsequent sections.
+* *V*(*0*) is the zero value of `x`'s [=effective-value-type=].
+
+<table class='data'>
+  <caption>Rules for joining multiple assignments to a function-scope variable.
+  <thead>
+    <tr><th>Statement
+        <th>Result
+        <th>Edges from the Result
+  </thead>
+  <tr><td class='nowrap'>
+          var *x*;
+      <td>*Vin*(*next*)
+      <td>*V*(*0*)
+  <tr><td class='nowrap'>
+          var *x* = *e*;<br>
+      <td rowspan=3>*Vin*(*next*)
+      <td rowspan=3>*V*(*e*)
+
+          Note: This is a [=full assignment=] to *x*.
+  <tr><td class='nowrap'>
+          *x* = *e*;<br>
+  <tr><td class='nowrap'>
+          **p* = *e*;<br>
+          where *p* is a [=let-declaration=] initialized to &*x*
+  <tr><td class='nowrap'>
+          *f*(*x*) = *e*;<br>
+      <td rowspan=3>*Vout*(*s*)
+      <td rowspan=3>*V*(*e*), *V*(*prev*)
+
+          Note: This is a [=partial assignment=] to *x*.
+
+          Note: Partial assignments include the previous value since only a subset of components are updated.
+  <tr><td class='nowrap'>
+          **p* = *e*;<br>
+          where *p* is a [=let-declaration=] initialized to &*f*(*x*)<br>
+  <tr><td class='nowrap'>
+          **f*(*p*) = *e*;<br>
+          where *p* is a [=let-declaration=] initialized to &*f*(*x*) or &*x*
+  <tr><td>*s1* *s2*<br>
+          where *Next* is in behavior of *s1*.
+
+          Note: *s1* often ends in a semicolon.
+      <td>*Vin*(*s2*)
+      <td>*Vout*(*s1*)
+  <tr><td class='nowrap'>
+          if *e* *s1* else *s2*<br>
+          where *Next* is in the behaviors of both *s1* and *s2*
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s1*), *Vout*(*s2*)
+  <tr><td class='nowrap'>
+          if *e* *s1* else *s2*<br>
+          where *Next* is in the behavior of *s1*, but not *s2*
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s1*)
+  <tr><td class='nowrap'>if *e* *s1* else *s2*<br>
+          where *Next* is in the behavior of *s2*, but not *s1*
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s2*)
+  <tr><td>loop { *s1* continuing { *s2* } }
+      <td>*Vin*(*s1*)
+      <td>*Vout*(*prev*), *Vout*(*s2*)
+  <tr><td>loop { *s1* continuing { *s2* } }
+      <td>*Vin*(*s2*)
+      <td>*Vout*(*s1*),<br>
+          *Vout*(*s*<sub>i</sub>)<br>
+          for all *s*<sub>i</sub> in *s1* whose behavior is {*Continue*} and transfer control to *s2*
+  <tr><td>loop { *s1* continuing { *s2* } }
+      <td>*Vin*(*next*)
+      <td>*Vout*(*s2*),<br>
+          *Vout*(*s*<sub>i</sub>)<br>
+          for all *s*<sub>i</sub> in *s1* whose behavior is {*Break*} and transfer control to *next*
```

Comment:
s2 can have include behaviour Break:


```
   loop {    { x =1;  break if true; } }
```

```

Reply:
Maybe it's subtle, but the break if statement must be the last statement in s2, so it is captured by the *Vout(s2)*.

Reply:
Ah, got it.
Nice.